include/TFEL/Material/Eshelby.hxx

```diff
@@ -0,0 +1,108 @@
+ * reserved.
+ * This project is publicly released under either the GNU GPL Licence
+ * or the CECILL-A licence. A copy of thoses licences are delivered
+ * with the sources of TFEL. CEA or EDF may also distribute this
+ * project under specific licensing conditions.
+ */
+
+#ifndef LIB_TFEL_MATERIAL_ESHELBY_HXX
+#define LIB_TFEL_MATERIAL_ESHELBY_HXX
+
+#include "TFEL/Math/st2tost2.hxx"
+#include "TFEL/Material/StiffnessTensor.hxx"
+
+namespace tfel::material
+{
+  /*!
+   * This function builds the Eshelby tensor for an ellipsoid whose semi-axis lengths are a,b,c
+   * embedded in an isotropic matrix (young,nu). 
+   * It returns an object of type st2tost2, which is the fourth-order Eshelby tensor, in a basis which is adapted to the ellipsoid :
+   * see Eshelby.ixx for the details. The basis IS NOT RELATED to the given order of the values a,b,c
+   * These expressions can be found in Torquato 2002 Random Heterogeneous Materials for the axisymetric ellipsoid
+   * and in Eshelby 1957 for other cases.
+   * \tparam    real: underlying type
+   * param 1 : nu
+   * param 2 : semi-axis of ellipsoid in direction 1
+   * param 3 :semi-axis of ellipsoid in direction 2
+   * param 4 : semi-axis of ellipsoid in direction 3
+   */
+     template <typename real>
+  TFEL_HOST_DEVICE const tfel::math::st2tost2<3u,real> EshelbyTensor(
+      const real&,
+      const real&,
+      const real&,
+      const real&);
+      
+  /*!
+   * This function builds the localisation tensor for an ellipsoid whose semi-axis lengths are a,b,c,
+   * whose elasticity tensor is C_i, 
+   * and embedded in an isotropic matrix (young,nu).
+   * The localisation tensor A is defined as follows : eps=A:eps0 where eps0 is the uniform strain tensor imposed at infinity,
+   * and eps is the strain tensor solution of Eshelby problem for the ellipsoid.
+   * The ellipsoid also has a specific orientation given by the vectors n_a, n_b.
+   * \returns an object of type st2tost2, which is the fourth-order localisation tensor A.
+   * \tparam    real: underlying type
+   * param 1,2 : young,nu : parameters of the isotropic matrix
+   * param 3 : st2tost2<3u,real> C_i : elastic tensor of the ellipsoid (may be anisotropic).
+   * param 4 : n_a : vector<real>,  direction of the principal axis whose length is a
+   * param 5 : a : length of semi-axis relative to the direction n_a
+   * param 6: n_b : vector<real>,  direction of the principal axis whose length is b
+   * param 7 : b : length of semi-axis relative to the direction n_b
+   * param 8 : c : length of the remaining semi-axis
+   */
+     template <typename real, typename StressType>
+  TFEL_HOST_DEVICE const tfel::math::st2tost2<3u,real> GeneralEllipsoidLocalisationTensor(
+      const StressType&,
+      const real&,
+      const tfel::math::st2tost2<3u, StressType>&,
+      const std::vector<real>&,
+      const real&,
+      const std::vector<real>&,
+      const real&,
+      const real&);
```

Comment:
Ok for stress, but then we shall have length type, no ?

Reply:
done